Automattic/sensei
# Add a script to add automatically the dist references to the pot file

Fixes #3422

### Changes proposed in this Pull Request

* Add a script to add automatically the dist references to the pot file.
  * It introduces 2 main scripts:
    * A chunk map generator, as a Webpack plugin. So when we build the project, it automatically generates a chunk map file. - Inspired by: https://github.com/Automattic/wp-calypso/blob/master/client/webpack/generate-chunks-map-plugin.js
    * A complete refactor in the `pot-dist-references.js` to add the dist references to the pot file, based on the generated chunk map file.

As now we have the chunk map dependency to generate the pot file, we added the `build` script as a first task in the `i18n:build`, so it'll also generate the build every time we generate the pot file.

### Testing instructions

* Run `npm run i18n:build`
* Check if the `sensei-lms.pot` was generated correctly (including the dist references to the JS files).

#### Testing the translations running in dev env:

* Go to the lang folder `cd lang`.
* Create a translation `po` file with poedit or running `msginit -l pt_BR -o sensei-lms-pt_BR.po`.
* If you generated through the command line, write some translations in the generated file for JS and PHP files.
* Run `msgfmt -o sensei-lms-pt_BR.mo sensei-lms-pt_BR.po` to generate the mo file.
* Run `wp i18n make-json . --no-purge` to generate the Jed file for JS messages.
* Move generated `.mo` and `.json` files to `/wp-content/languages/plugins`.
* Go to WP-admin > Settings > General and update the language.
* Access the JS page that contains the messages you translated to check if the messages are translated.
* Access the PHP page to check if the messages are translated.

### References

* PR that introduced the i18n scripts: https://github.com/Automattic/sensei/pull/3219
